cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -95,6 +99,33 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 			NCVersion: int(nc.Version),
 		}
 	}
+
+	// Add IPs from CIDR block to the secondary IPConfigs
+	if nc.Type == v1alpha.VNETBlock {
+		// Delete the IPs upto(including) first IP of primary prefix
+		startAddr := primaryPrefix.Masked().Addr()
+		endAddr := primaryPrefix.Addr()
+		for addr := startAddr; addr != endAddr.Next() && primaryPrefix.Contains(addr); addr = addr.Next() {
```

Comment:
Why is there a loop here as it will only be one IP that needs to be removed always and secondaryIPConfigs is a map where you can directly access it using the address string.

Reply:
ok let me explain my understanding with an example.
Lets say PrimaryIP is 10.224.0.4/27. 
With code above this if block(which is same for overlay), we add IPs starting from 10.224.0.0(not 10.224.0.4) to 10.224.0.31 to secondaryIP config.
Now we need to remove not only 10.224.0.4 but  10.224.0.0, 10.224.0.1, 10.224.0.2 &  10.224.0.3 as well.

Reply:
ok, updated